card-game-client/app/components/ui/game-card.tsx

```diff
@@ -1,376 +1,394 @@
-    e: React.ChangeEvent<HTMLInputElement>,
-    index: number
-  ) => {
-    const newOptions = [...editedOptions];
-    newOptions[index] = e.target.value;
-    setEditedOptions(newOptions);
-  };
+    const handleOptionChange = (
+        e: React.ChangeEvent<HTMLInputElement>,
+        index: number
+    ) => {
+        const newOptions = [...editedOptions];
+        newOptions[index] = e.target.value;
+        setEditedOptions(newOptions);
+    };
 
-  const handleOptionBlur = () => {
-    setEditingOption(null);
-  };
+    const handleOptionBlur = () => {
+        setEditingOption(null);
+    };
 
-  const handleNextClick = () => {
-    if (currentItemIndex < (subCategoryItems?.length ?? 0) - 1) {
-      setCurrentItemIndex(currentItemIndex + 1);
-      setSelectedOption(null);
-      setTimeLeft(timeLimit);
-    } else {
-      onClose?.();
-    }
-  };
+    const handleNextClick = () => {
+        if (currentItemIndex < (subCategoryItems?.length ?? 0) - 1) {
+            setCurrentItemIndex(currentItemIndex + 1);
+            setSelectedOption(null);
+            setTimeLeft(timeLimit);
+        } else {
+            onClose?.();
+        }
+    };
 
-  const handleSave = () => {
-    const payload = {
-      subCategory: editedTitle,
-      category: editedCategory,
-      question: editedQuestion,
-      answer: editedAnswer,
-      answerOption: editedOptions.toString(),
-      file: uploadFile,
+    const handleEdit = () => {
+        setEditing(true);
     };
 
-    axios({
-      method: "post",
-      url: "/card/insert",
-      data: payload,
-      headers: { "Content-Type": "multipart/form-data" },
-    })
-      .then((response) => {
-        setAddedCardList([...addedCardList, response.data]);
-      })
-      .catch((error) => {
-        console.error("There was an error!", error);
-      })
-      .finally(() => {
-        onClose?.();
-      });
-  };
+    const handleCancelEdit = () => {
+        setEditing(false);
+        setEditedTitle(title || "");
+        setEditedQuestion(question || "");
+        setEditedCategory(category || "");
+        setEditedAnswer(answer || "");
+        setEditedOptions(options || ["", "", "", ""]);
+    };
 
-  const handleImageUpload = (e: React.ChangeEvent<HTMLInputElement>) => {
-    const file = e.target.files?.[0];
-    if (file) {
-      const reader = new FileReader();
-      reader.onloadend = () => {
-        setImageUrl(reader.result as string);
-        setUploadFile(file);
-      };
-      reader.readAsDataURL(file);
-    }
-  };
+    const handleSave = () => {
+        const payload = {
+            subCategory: editedTitle,
+            category: editedCategory,
+            question: editedQuestion,
+            answer: editedAnswer,
+            answerOption: editedOptions.toString(),
+            file: uploadFile,
+        };
 
-  const currentItem = subCategoryItems?.[currentItemIndex];
+        axios({
+            method: editing ? "put" : "post",
```

Comment:
method should be just "post"
In backend, there are only two methods that we utilized